src/wallet/rpcdump.cpp

```diff
@@ -510,3 +534,110 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
     file.close();
     return NullUniValue;
 }
+
+
+UniValue importmulti(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 2)
+        throw runtime_error(
+            "importmulti '[{\"type\":\"privkey\",\"value\":\"mkjjX...\"},...]' (rescan) \n\n"
+            "Import several types of addresses (private and public keys, transaction addresses/scripts) with only one rescan\n"
+
+            "Arguments:\n"
+            "1. json request array     (json, required) Data to be imported\n"
+            "  [     (json array of json objects)\n"
+            "    {\n"
+            "      \"type\": \"privkey | pubkey | address\", (string, required) Type of address\n"
+            "      \"value\": \"...\",                       (string, required) Value of the address\n"
+            "      \"timestamp\": 1454686740,                (integer, optional) Timestamp\n"
+            "      \"label\": \"...\"                        (string, optional) Label\n"
+            "      \"p2sh\": true | false                    (bool, optional, default=false) Value is a P2SH\n"
+            "    }\n"
+            "  ,...\n"
+            "  ]\n"
+            "2. rescan                 (boolean, optional, default=true)\n"
+
+            "\nExamples:\n" +
+            HelpExampleCli("importmulti", "'[ { \"type\":\"privkey\", \"value\":\"<my private key>\", \"timestamp\":1455191478 },"
+                                          "{ \"type\":\"pubkey\", \"value\":\"<my public key>\", \"label\":\"example 1\", \"timestamp\":1455191480 } ]' true") +
+            HelpExampleCli("importmulti", "'[{ \"type\":\"pubkey\", \"value\":\"<my public key>\", \"label\":\"example 1\", \"timestamp\":1455191464 } ]' false") + HelpExampleRpc("importmulti", "[ { \"type\":\"privkey\", \"value\":\"<my private key>\" },"
+                                                                                                                                                                                                  "{ \"type\":\"pubkey\", \"value\":\"<my public key>\", \"label\":\"example 1\", \"timestamp\":1455191480 } ], true") +
+            HelpExampleRpc("importmulti", "'[{ \"type\":\"pubkey\", \"value\":\"<my public key>\", \"label\":\"example 1\", \"timestamp\":1455191464 } ]', false") +
+
+            "\nResponse is an array with the same size as the input that has the execution result :\n"
+            "  [ { \"result\": true | false } , ... ]\n");
+
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (params.size() == 1)
+        RPCTypeCheck(params, boost::assign::list_of(UniValue::VARR));
+    else
+        RPCTypeCheck(params, boost::assign::list_of(UniValue::VARR)(UniValue::VBOOL));
+
+    const UniValue& jRequest = params[0].get_array();
```

Comment:
No need to call `get_array` as this is just for type check, which is done in `RPCTypeCheck` before.